compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt

```diff
@@ -43,26 +43,70 @@ import java.awt.event.*
 import java.awt.event.KeyEvent
 import java.awt.im.InputMethodRequests
 import javax.accessibility.Accessible
-import javax.swing.SwingUtilities
+import javax.swing.JComponent
 import kotlin.coroutines.AbstractCoroutineContextElement
 import kotlin.coroutines.CoroutineContext
 import kotlinx.coroutines.CoroutineExceptionHandler
 import org.jetbrains.skia.Canvas
 import org.jetbrains.skiko.*
-import org.jetbrains.skiko.SkiaLayer
 
-internal class ComposeLayer(
-    private val skiaLayerAnalytics: SkiaLayerAnalytics
-) {
+/**
+ * Provides a base implementation for integrating a Compose scene with AWT/Swing.
+ * It allows setting Compose content by [setContent], this content should be drawn on [component].
+ *
+ * Inheritors should call [attachComposeToComponent], so events that came to [component] will be transferred to [ComposeScene]
+ */
+internal abstract class ComposeLayer<out T : JComponent> {
```

Comment:
I don't really like this generic parameter. But I use it to easily provide different implementations in ComposePanel, WDYT?

Reply:
I don't think it's particularly bad, but I also don't think it's really needed.
- Remove the type argument
    - Change the type of `component` to `JComponent` in `ComposeLayer`.
    - Change the type of `component` to `SkiaSwingLayerComponent` in `ComposeSwingLayer`
    - Change the type of `component` to `SkiaLayer` in `ComposeWindowLayer`.

It seems to compile for me.

Reply:
You are right. Did it